python/link_prediction/link_prediction.py

```diff
@@ -0,0 +1,167 @@
+        try:
+            link_prediction_parameters.__setattr__(key, value)
+        except Exception as exception:
+            return mgp.Record(status=0, message=repr(exception))
+
+    return mgp.Record(status=1, message="OK")
+
+
+@mgp.read_proc
+def train_eval(ctx: mgp.ProcCtx) -> mgp.Record(auc_score=float):
+    """
+    """
+    graph, new_to_old = _get_dgl_graph_data(ctx)  # dgl representation of the graph and dict new to old index
+    if test_conversion(graph=graph, new_to_old=new_to_old, ctx=ctx) is False:
+        print("Remapping failed")
+
+    # Train g is a graph which has removed test edges
+    # Others are positive and negative train and test graphs
+
+    # NOTE: THIS LINE BELOW RESULTS IN RAM OVERFLOW
+    # train_g, train_pos_g, train_neg_g, test_pos_g, test_neg_g = link_prediction_util.preprocess(g)
+    # Create a model and edge predictor
+    # Send to the GraphSAGE model features
+    # print(train_g.ndata)
+    # model = GraphSAGE(train_g.ndata['features'].shape[1], 16)
+    # pred = DotPredictor()
+
+    # link_prediction_util.train(model, pred, train_g, train_pos_g, train_neg_g)
+    # trained_hidden_features = model(train_g, train_g.ndata["features"])  # ndata returns a node-data view for getting/setting node features. This will return node feature fear = (Tensor) or dict(str, Tensor) if dealing
+    # auc_score = link_prediction_util.test(trained_hidden_features, pred, test_pos_g, test_neg_g)
+
+    # Return just some generic result
+    rec = mgp.Record(auc_score=0.01)
+    return rec
+
+
+@mgp.read_proc
+def predict_link_score(ctx: mgp.ProcCtx, src_vertex: mgp.Vertex, dest_vertex: mgp.Vertex) -> mgp.Record(score=mgp.Number):
+    """
+    :param ctx: Reference to the execution context.
+    :param src_vertex: Source vertex
+    :param dest_vertex: Destination vertex.
+    :return: Score between 0 and 1. 
+    """
+    result = mgp.Record(score=0.61)
+    return result
+
+
+##############################
+# Convert to DGL graph, consider extracting such methods to another file.
+##############################
+
+
+def _get_dgl_graph_data(ctx: mgp.ProcCtx) -> Tuple[dgl.graph, Dict[int32, int32]]:
+    """
+    Creates dgl representation of the graph.
+    :param ctx: Reference to the context execution.
+    :return: Tuple of DGL graph representation and dictionary of mapping new to old index.
+    """
+    src_nodes, dest_nodes = [], []  # for saving the edges
+
+    new_to_old = dict()  # map of new node index to old node index
+    old_to_new = dict()  # map of old node index to new node index
+    features = []  # map of new node index to its feature
+    ind = 0
+
+    for vertex in ctx.graph.vertices:
+        for edge in vertex.out_edges:
+            src_node, dest_node = edge.from_vertex, edge.to_vertex
+            src_id_old = int(src_node.properties.get("id"))
+            src_features = src_node.properties.get("features")
```

Comment:
Check whether "features" exist when calling, and make them as argument of input function